github/search.go

```diff
@@ -65,17 +65,16 @@ type CommitsSearchResult struct {
 
 // CommitResult represents a commit object as returned in commit search endpoint response.
 type CommitResult struct {
-	Hash           *string     `json:"hash,omitempty"`
-	Message        *string     `json:"message,omitempty"`
-	AuthorID       *int        `json:"author_id,omitempty"`
-	AuthorName     *string     `json:"author_name,omitempty"`
-	AuthorEmail    *string     `json:"author_email,omitempty"`
-	AuthorDate     *Timestamp  `json:"author_date,omitempty"`
-	CommitterID    *int        `json:"committer_id,omitempty"`
-	CommitterName  *string     `json:"committer_name,omitempty"`
-	CommitterEmail *string     `json:"committer_email,omitempty"`
-	CommitterDate  *Timestamp  `json:"committer_date,omitempty"`
-	Repository     *Repository `json:"repository,omitempty"`
+	SHA         *string     `json:"sha,omitempty"`
+	Commit      *Commit     `json:"commit,omitempty"`
+	Author      *User       `json:"author,omitempty"`
+	Committer   *User       `json:"committer,omitempty"`
+	Parents     []Commit    `json:"parents,omitempty"`
```

Comment:
Based on #180 and https://github.com/google/go-github/pull/520#discussion_r98745029 (/cc @gmlewis), I think we'd want to go with `[]*Commit` here. Even though it's a little inconsistent with other similar structs.

At least that's what I think @gmlewis would request.